library/manageiq_alert.py

```diff
@@ -0,0 +1,272 @@
+#!/usr/bin/python
+
+
+DOCUMENTATION = '''
+---
+module: manageiq_alert
+description: The manageiq_alert module supports adding, updating and deleting alert definitions in ManageIQ.
+short_description: management of alerts in ManageIQ
+requirements: [ ManageIQ/manageiq-api-client-python ]
+author: Daniel Korn (@dkorn)
+options:
+  miq_url:
+    description:
+      - the manageiq environment url
+    default: MIQ_URL env var if set. otherwise, it is required to pass it
+  miq_username:
+    description:
+      - manageiq username
+    default: MIQ_USERNAME env var if set. otherwise, it is required to pass it
+  miq_password:
+    description:
+      - manageiq password
+    default: MIQ_PASSWORD env var if set. otherwise, it is required to pass it
+  description:
+    description:
+      - the alert definition description in manageiq
+    required: false
+    default: null
+  db:
+    description:
+      - the db entry of the entity to base the alert on in manageiq
```

Comment:
I don't quite like the name `db`, nor the fact it takes internal class names as values.
How about something like `resource` or `entity`?  And perhaps a mapping hash like you did in a couple places.